### PR TITLE
Fix matchProvidersWithUser inconsistent

### PR DESCRIPTION
**Pull request by auxten**

### CODE REVIEW

File: blockproducer/metastate.go [hunk 1]
```diff
@@ -763,6 +764,41 @@ func (s *metaState) matchProvidersWithUser(tx *types.CreateDatabase) (err error)
...
+	minerCount int) (
+	m MinerInfos, err error,
+) {
+	// create new merged map
+	allProviderMap := make(map[proto.AccountAddress]*types.ProviderProfile)
+	for k, v := range s.readonly.provider {
+		allProviderMap[k] = v
+	}
+	for k, v := range s.dirty.provider {
+		allProviderMap[k] = v
```
Comment (leventeliu) [inline blockproducer/metastate.go]: Branch here, delete from `allProviderMap` if `v == nil`.

File: blockproducer/metastate_test.go [hunk 2]
```diff
@@ -31,6 +31,8 @@ import (
 	"github.com/CovenantSQL/CovenantSQL/proto"
 	"github.com/CovenantSQL/CovenantSQL/route"
 	"github.com/CovenantSQL/CovenantSQL/types"
+	"github.com/CovenantSQL/CovenantSQL/utils/log"
+
```
Comment (leventeliu) [inline blockproducer/metastate_test.go:35]: Delete blank line.

Comment (auxten) [reply blockproducer/metastate_test.go:35]: My Goland format stuff

File: blockproducer/metastate.go [hunk 3]
```diff
@@ -756,13 +757,51 @@ func (s *metaState) matchProvidersWithUser(tx *types.CreateDatabase) (err error)
...
+	// create new merged map
+	allProviderMap := make(map[proto.AccountAddress]*types.ProviderProfile)
+	for k, v := range s.readonly.provider {
+		allProviderMap[k] = v
+	}
+	for k, v := range s.dirty.provider {
+		if v == nil {
+			delete(allProviderMap, k)
+		}
+		allProviderMap[k] = v
```
Comment (leventeliu) [inline blockproducer/metastate.go]: Shouldn't it be in `else`?

**Review by leventeliu — APPROVED** [review]
LGTM.

**Review by xq262144 — APPROVED** [review]
Approve

**Pull request merged**